packages/flutter/lib/src/rendering/slivers.dart

```diff
@@ -0,0 +1,1850 @@
+        switch (axis) {
+          case Axis.vertical:
+            size = new Size(constraints.maxWidth, 0.0);
+            break;
+          case Axis.horizontal:
+            size = new Size(0.0, constraints.maxHeight);
+            break;
+        }
+        offset.applyViewportDimension(0.0);
+      }
+      offset.applyContentDimensions(0.0, 0.0);
+      return;
+    }
+    assert(center.parent == this);
+
+    double extent;
+    double crossExtent;
+    if (shrinkWrap) {
+      assert(constraints.hasBoundedHeight && constraints.hasBoundedWidth);
+      switch (axis) {
+        case Axis.vertical:
+          extent = constraints.maxHeight;
+          crossExtent = constraints.maxWidth;
+          break;
+        case Axis.horizontal:
+          extent = constraints.maxWidth;
+          crossExtent = constraints.maxHeight;
+          break;
+      }
+    } else {
+      switch (axis) {
+        case Axis.vertical:
+          extent = size.height;
+          crossExtent = size.width;
+          break;
+        case Axis.horizontal:
+          extent = size.width;
+          crossExtent = size.height;
+          break;
+      }
+    }
+
+    final double centerOffsetAdjustment = center.centerOffsetAdjustment;
+
+    double correction;
+    double effectiveExtent;
+    do {
+      assert(offset.pixels != null);
+      correction = _attemptLayout(extent, crossExtent, offset.pixels + centerOffsetAdjustment);
+      if (correction != 0.0) {
+        offset.correctBy(correction);
+      } else {
+        if (shrinkWrap) {
+          switch (axis) {
+            case Axis.vertical:
+              effectiveExtent = constraints.constrainHeight(_shrinkWrapExtent);
+              break;
+            case Axis.horizontal:
+              effectiveExtent = constraints.constrainWidth(_shrinkWrapExtent);
+              break;
+          }
+          offset.applyViewportDimension(effectiveExtent);
+        } else {
+          switch (axis) {
+            case Axis.vertical:
+              effectiveExtent = size.height;
+              break;
+            case Axis.horizontal:
+              effectiveExtent = size.width;
+              break;
+          }
+        }
+        if (offset.applyContentDimensions(
+              math.min(0.0, _minScrollExtent + effectiveExtent * anchor),
+              math.max(0.0, _maxScrollExtent - effectiveExtent * (1.0 - anchor)
+           )))
+          break;
+      }
+    } while (true);
+
+    assert(shrinkWrap != sizedByParent);
+    if (shrinkWrap) {
+      switch (axis) {
+        case Axis.vertical:
+          size = new Size(crossExtent, effectiveExtent);
+          break;
+        case Axis.horizontal:
+          size = new Size(effectiveExtent, crossExtent);
+          break;
+      }
+    }
+  }
+
+  double _attemptLayout(double extent, double crossExtent, double correctedOffset) {
+    double result;
+
+    _minScrollExtent = 0.0;
+    _maxScrollExtent = 0.0;
+    _shrinkWrapExtent = 0.0;
```

Comment:
Yeah. I could also pass a struct around and have it be filled in then pull the values at the end and put them into the fields, but it seemed unnecessary since it's all private anyway. Happy to change it if you prefer it that way though.